pkg/types/telemetrytypes/field.go

```diff
@@ -137,7 +137,7 @@ type FieldValueSelector struct {
 	Limit         int    `json:"limit"`
 }
 
-func DataTypeCollisionHandledFieldName(key *TelemetryFieldKey, value any, tblFieldName string) (string, any) {
+func DataTypeCollisionHandledFieldName(key *TelemetryFieldKey, value any, tblFieldName string, isComparisonOperator bool) (string, any) {
```

Comment:
can we pass the operator directly and use `IsComparisonOperator` inside `DataTypeCollisionHandledFieldName` instead. It will give `DataTypeCollisionHandledFieldName` access to the exact operator used and any functions available.

Reply:
Passing the operator results in import cycle, so I used this method.

Reply:
Should the function `DataTypeCollisionHandledFieldName` even be here in that case; We can move to https://github.com/SigNoz/signoz/tree/main/pkg/querybuilder?